### PR TITLE
MediaSession API support

### PR DESCRIPTION
**Pull request by LofhJann**
MediaSession typing file, Keyboard support for multimedia keys and metadata in media notifications.

As requested in issue #296

### CODE REVIEW

File: src/app/components/Common/Player/Player.tsx [hunk 1]
```diff
@@ -139,6 +143,21 @@ class Player extends React.Component<PlayerProps> {
       <span className={cx(styles.albumName)}>{nowPlayingItem.attributes.albumName}</span>
     );
 
+    if (navigator.mediaSession && hasMeta) {
```
Comment (raphaelvigee) [inline src/app/components/Common/Player/Player.tsx]: This shouldn't be part of the render, it should be encapsulated in a `useEffect` depending on `[nowPlayingItem]`

File: src/app/types/MediaSession.d.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  public title: string;
+  // Media's artist.
+  public artist: string;
+  // Media's album.
+  public album: string;
+  // Media's artwork.
+  public artwork: MediaImage[];
+
+  constructor(init?: MediaMetadataInit);
+  // Media's title.
```
Comment (raphaelvigee) [inline src/app/types/MediaSession.d.ts]: This should be on line 55

File: src/app/components/Common/Player/Player.tsx [hunk 3]
```diff
@@ -48,6 +48,10 @@ class Player extends React.Component<PlayerProps> {
     Mousetrap.bind('left', this.handlePrevious, 'keyup');
     Mousetrap.bind('right', this.handleNext, 'keyup');
     Mousetrap.bind('space', togglePlayback, 'keyup');
+    if (navigator.mediaSession) {
+      navigator.mediaSession.setActionHandler('nexttrack', this.handleNext);
```
Comment (raphaelvigee) [inline src/app/components/Common/Player/Player.tsx]: We should hook all available action handlers: `play`, `pause`, `seekbackward`, `seekforward`, `previoustrack` and `nexttrack`

**Review by raphaelvigee — CHANGES_REQUESTED** [review]

**Comment by raphaelvigee** [comment]
I think the `Player` is not the right place for that, a new `MediaSessionProvider` (or similar) should be created  to handle all interactions  with the  `MediaSession`

File: src/app/types/MediaSession.d.ts (hunk 2 above)
Comment (LofhJann) [reply src/app/types/MediaSession.d.ts]: Which line you mean?

Comment (raphaelvigee) [reply src/app/types/MediaSession.d.ts]: This  comment  is  related to line  55

**Comment by raphaelvigee** [comment]
Maybe I haven't been clear  with  my thoughts about the `MediaSessionProvider`.
I think a new  component  called `MediaSessionProvider` should  be created containing all the lifecycle  related to `MediaSession` and mounted into the  tree  in `App`. `Player` should  not be touched.

File: src/app/components/Common/Player/Player.tsx (hunk 1 above)
Comment (LofhJann) [reply src/app/components/Common/Player/Player.tsx]: Never really used useEffect hooks before, understand the principle but not quite sure how to incorporate it here. May you provide me with some example on how it should be used here? Will gladly refactor it, seems like better way for sure

**Comment by JuhQ** [comment]
This is awesome! 👍

**Comment by jordan-heemskerk** [comment]
I'd love to help push this along. @LofhJann @raphaelvigee let me know if there's anything I can do.

**Comment by raphaelvigee** [comment]
@jordan-heemskerk  You can have a look at my [comment](https://github.com/Musish/Musish/pull/446#issuecomment-526334546) and reviews on this PR

**Pull request closed without merging**